crypto/cipher/aes_icm.c

```diff
@@ -92,7 +92,7 @@ srtp_debug_module_t srtp_mod_aes_icm = {
  *
  */
 
-static srtp_err_status_t srtp_aes_icm_alloc_ismacryp (srtp_cipher_t **c, int key_len, int forIsmacryp)
+static srtp_err_status_t srtp_aes_icm_alloc (srtp_cipher_t **c, int key_len)
```

Comment:
This function signature change may cause problems.

In `aes_icm.c:494` it is used like so:

~~~.c
const srtp_cipher_type_t srtp_aes_icm = {
    srtp_aes_icm_alloc, // throws error
    srtp_aes_icm_dealloc,
    srtp_aes_icm_context_init,
    0,                          /* set_aad */
    srtp_aes_icm_encrypt, // throws error
    srtp_aes_icm_encrypt, // throws error
    srtp_aes_icm_set_iv,
    0,                          /* get_tag */
    srtp_aes_icm_description,
    &srtp_aes_icm_test_case_1,
    SRTP_AES_ICM
};
~~~

In 'cipher.h:142' the `srtp_cipher_type_t` is defined as:

~~~.c
typedef struct srtp_cipher_type_t {
    srtp_cipher_alloc_func_t alloc;
    srtp_cipher_dealloc_func_t dealloc;
    srtp_cipher_init_func_t init;
    srtp_cipher_set_aad_func_t set_aad;
    srtp_cipher_encrypt_func_t encrypt;
    srtp_cipher_encrypt_func_t decrypt;
    srtp_cipher_set_iv_func_t set_iv;
    srtp_cipher_get_tag_func_t get_tag;
    const char                       *description;
    const srtp_cipher_test_case_t         *test_data;
    srtp_cipher_type_id_t id;
} srtp_cipher_type_t;
~~~

Where the signature of `srtp_cipher_alloc_func_t` is:
~~~.c
typedef srtp_err_status_t (*srtp_cipher_alloc_func_t)
    (srtp_cipher_pointer_t *cp, int key_len, int tag_len);
~~~

Since `tag_len` does not seem to make sense in the context of `srtp_aes_icm_alloc_ismacryp(...)` I suggest adding a dummy `int` in order to get the signature to match? Not sure what the correct approach here is.